modules/Bio/EnsEMBL/Variation/VCFCollection.pm

```diff
@@ -486,59 +486,77 @@ sub get_all_VariationFeatures_by_Slice {
+      my @info_format = split('\|', $description[1]);
+      foreach my $vf (@vfs) { 
+        if ( $vcf_info_field ){
+          my $info = $vf->{vcf_record}->get_info;
+          my @vcf_variants = $info->{$vcf_info_field};
+          foreach my $vcf_variant ( @vcf_variants ) {
+            if ( defined($vcf_variant) ){
+              my @transcript_vars = split(',',$vcf_variant);
+              foreach my $transcript_var ( @transcript_vars ){
+                my @transcript_split = split('\|',$transcript_var);
+                my %transcript_map = zip @info_format, @transcript_split;
+                next if not defined($transcript_map{'Feature_type'});
+                if ( $transcript_map{'Feature_type'} eq 'Transcript') {
+                  my $tv = Bio::EnsEMBL::Variation::TranscriptVariation->new_fast({
+                    variation_feature  => $vf,
+                    _feature_stable_id => $transcript_map{'Feature'}
+                  });  
+                  #Add variation Feature seq from transcript map
+                  my $tva = Bio::EnsEMBL::Variation::TranscriptVariationAllele->new_fast({
+                    variation_feature_seq => $transcript_split[0],
+                    is_reference               => 0
+                  });
+                  my @cons_list = split('&',$transcript_map{'Consequence'});
+                  foreach my $con (@cons_list){
+                    if ( exists $OVERLAP_CONSEQUENCES{$con} ) {
+                      my $new_cons = $OVERLAP_CONSEQUENCES{$con};
+                      $tva->add_OverlapConsequence($new_cons);
+                    }
+                    else{
+                      print("The consequence is not available:",$con,"\n");
+                    }
                   }
+                  $tv->add_TranscriptVariationAllele($tva);
+                  $vf->add_TranscriptVariation($tv);
                 }
-                $tv->add_TranscriptVariationAllele($tva);
-                $vf->add_TranscriptVariation($tv);
               }
             }
           }
         }
+        else{ 
+          $vf->{intergenic_variation} = Bio::EnsEMBL::Variation::IntergenicVariation->new(
+            -variation_feature  => $vf,
+            -no_ref_check       => 1,
+          );
+          weaken($vf->{intergenic_variation}->{base_variation_feature});
+        }
+        $vf->_finish_annotation();
       }
-      else{ 
-        $vf->{intergenic_variation} = Bio::EnsEMBL::Variation::IntergenicVariation->new(
-          -variation_feature  => $vf,
-          -no_ref_check       => 1,
-        );
-        weaken($vf->{intergenic_variation}->{base_variation_feature});
+    } elsif ($vcf_info_field eq 'CLNSIG'){
+      foreach my $vf (@vfs) {
+        my $info = $vf->{vcf_record}->get_info;
+        my $vcf_variants = $info->{$vcf_info_field};
+        my @clnsig_vars = split('\|',$vcf_variants);
+        foreach my $clnsig_var (@clnsig_vars) {
+          $clnsig_var =~ s/_/ /g;
```

Comment:
According to JIRA - https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-5046 we have some specific format for ClinVar clinical significance. We also put them out in our doc like this - https://www.ensembl.org/info/genome/variation/phenotype/phenotype_annotation.html#clin_significance . So, we should add the clinical significance term according to this list.

For example, It would be better to have some more filtering. Like -
```
# removing leading underscore _
_risk_factor -> risk factor
# splitting on / and further processing
Pathogenic/Likely_pathogenic -> we will get pathogenic and likely pathogenic
```
I found these two cases from the values from chromosome 1 of the ClinVar VCF. You may find some more.

And, maybe you should filter out the value also so that it matches exactly with our attribs here - https://github.com/Ensembl/ensembl-variation/blob/main/modules/Bio/EnsEMBL/Variation/Utils/Config.pm#L68
You can import the array in this file and do a grep before adding them to the `$vf->{clinical_significance}`

Reply:
I'm adding line to trim leading and trailing whitespaces from the string: `$clnsig_lc =~ s/^\s+|\s+$//;`.
For the splitting on "/" I have included the forward slash as another character by which to split by - since the split happens before any processing takes place. I've tested and the example "benign/likely_benign" will come out as two entries of "benign" and "likely benign"